src/CommunityToolkit.Maui/Views/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,238 @@
+
+	/// <summary>The backing store for the <see cref="Email" /> bindable property.</summary>
+	public static readonly BindableProperty? EmailProperty = BindableProperty.Create(nameof(Email), typeof(string), typeof(GravatarImageSource), defaultValue: GravatarImageSourceDefaults.DefaultEmail, propertyChanged: OnEmailPropertyChanged);
+
+	/// <summary>The backing store for the <see cref="Image" /> bindable property.</summary>
+	public static readonly BindableProperty ImageProperty = BindableProperty.Create(nameof(Image), typeof(DefaultImage), typeof(GravatarImageSource), defaultValue: GravatarImageSourceDefaults.Defaultimage, propertyChanged: OnDefaultImagePropertyChanged);
+
+	int gravatarSize;
+
+	/// <summary>Initializes a new instance of the <see cref="GravatarImageSource"/> class.</summary>
+	public GravatarImageSource()
+	{
+	}
+
+	/// <summary>On property changed.</summary>
+	/// <param name="propertyName">Property name.</param>
+#pragma warning disable CS8625 // Cannot convert null literal to non-nullable reference type. NOTE: This is to match override
+	protected override void OnPropertyChanged(string propertyName = null)
+#pragma warning restore CS8625 // Cannot convert null literal to non-nullable reference type. NOTE: This is to match override
+	{
+		if (propertyName is not null && (propertyName == CacheValidityProperty.PropertyName || propertyName == CachingEnabledProperty.PropertyName))
+		{
+			OnUriChanged();
+		}
+
+		base.OnPropertyChanged(propertyName);
+	}
+
+	/// <summary>Gets or sets a <see cref="TimeSpan"/> structure that indicates the length of time after which the image cache becomes invalid.</summary>
+	public TimeSpan CacheValidity
+	{
+		get => (TimeSpan)GetValue(CacheValidityProperty);
+		set => SetValue(CacheValidityProperty, value);
+	}
+
+	/// <summary>Gets or sets a Boolean value that indicates whether caching is enabled on this <see cref="GravatarImageSource"/> object.</summary>
+	public bool CachingEnabled
+	{
+		get => (bool)GetValue(CachingEnabledProperty);
+		set => SetValue(CachingEnabledProperty, value);
+	}
+
+	/// <summary>Gets or sets the email address.</summary>
+	public string? Email
+	{
+		get => (string)GetValue(EmailProperty);
+		set => SetValue(EmailProperty, value);
+	}
+
+	/// <summary>Gets or sets the default image.</summary>
+	public DefaultImage Image
+	{
+		get => (DefaultImage)GetValue(ImageProperty);
+		set => SetValue(ImageProperty, value);
+	}
+
+	/// <summary>Gets a value indicating whether the control URI is empty.</summary>
+	public override bool IsEmpty => Uri == null;
+
+	/// <summary>Gets or sets the URI for the image to get.</summary>
+	[System.ComponentModel.TypeConverter(typeof(UriTypeConverter))]
+	public Uri? Uri { get; private set; }
+
+	/// <summary>Gets or sets the control size property.</summary>
+	int GravatarSize
+	{
+		get => gravatarSize;
+		set
+		{
+			switch (value)
```

Comment:
Use Math.Clamp

Reply:
I honestly did not know about this API.  Thanks for the education, much appreciated!